packages/x-date-pickers/src/AdapterMoment/index.ts

```diff
@@ -62,7 +62,9 @@ export class AdapterMoment
       .map((token) => {
         const firstCharacter = token[0];
         if (firstCharacter === 'L' || firstCharacter === ';') {
-          return this.moment.localeData().longDateFormat(token as LongDateFormatKey);
+          return this.moment
+            .localeData(this.getCurrentLocaleCode())
+            .longDateFormat(token as LongDateFormatKey);
```

Comment:
This is needed because otherwise the default locale of moment will be applied (the last imported one)

It's a part of code copypasted from date-io. after this PR I will add the modification to date-io